### PR TITLE
CREST addition to bde workflow 

### PR DESCRIPTION
**Pull request by shreesowndarya**
I have done it in a new file because the way we write gaussian gif file changes from mol object to .xyz from CREST. I wasn't sure if this would become the standard procedure so I did it as a separate file.

Things done:

1. Adding in crest after MMFF optimization (mol object is converted to .xyz following which CREST is done)
2. based on best conformers (xyz) from CREST, the gjf file is written now from the crest_best.xyz file

I debugged independently the following:

1. mol object after MMFF optimization to xyz file for CREST
2. writing of .gjf file from the crest_best.xyz 

Checks needed:
How crest is invoked on eagle and if I set up the temporary directories correctly!

Once the .gjf file is written the protocol is the same afterwards.

### CODE REVIEW

File: bde/gaussian_crest.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+        self.run_hex = uuid.uuid4().hex[:6]
+        self.xyz = tmpdirname + '/{0}_{1}.xyz'.format(self.cid, self.run_hex)
+        self.crest_out = tmpdirname + '/{0}_{1}.crest.out'.format(self.cid, self.run_hex)
+
+        #writing to xyzfile
+        rdkit.Chem.rdmolfiles.MolToXYZFile(mol,self.xyz,confId=confId)
+
+        #running calc in temporary TemporaryDirectory : NEEDS CHECK
+        with tempfile.TemporaryDirectory(dir=tmpdirname) as cresttmp:
```
Comment (pstjohn) [inline bde/gaussian_crest.py]: I'm not sure this is actually doing anything? you could put the other output in cresttmp, or chdir into cresttmp, but this is all already happening inside a temporary directory (tmpdirname) so the nested temp directory (i.e., what I did for gaussian) is probably not necessary

File: bde/gaussian_crest.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+            env = os.environ.copy()
+            crest_cmd = "crest {0} > {1}".format(
+                self.xyz, self.crest_out)
+
+            subprocess.run(crest_cmd, shell=True, env=env,
+                           timeout=self.crest_timeout)
+
+            self.best_xyz = tmpdirname + '/{0}_{1}_best.xyz.'.format(self.cid, self.run_hex)
```
Comment (pstjohn) [inline bde/gaussian_crest.py]: this could probably get moved with the other path variables in lines 133-135

**Comment by pstjohn** [comment]
looks great! the install for CREST does seem a bit onerous, even if xtb is installable with conda
https://github.com/grimme-lab/crest

I wonder if their intel binaries would work on eagle

**Comment by shreesowndarya** [comment]
Should I contact HPC about how xtb/crest can be installed on eagle??

**Pull request merged**